internal/config/types.go

```diff
@@ -414,16 +417,63 @@ func (c *Config) AreReceiversConfigured() bool {
 		len(c.Collector.Receivers.TcplogReceivers) > 0
 }
 
-func isAllowedDir(dir string, allowedDirs []string) bool {
-	if !strings.HasSuffix(dir, "/") && filepath.Ext(dir) == "" {
-		dir += "/"
+// isAllowedDir checks if the given path is in the list of allowed directories.
+// It returns true if the path is allowed, false otherwise.
+// If the path is allowed but does not exist, it also logs a warning.
+// It also checks if the path is a file, in which case it checks the directory of the file.
+func isAllowedDir(path string, allowedDirs []string) bool {
```

Comment:
Can you pass the context into this function as well so that we can use it in the log statements?